### PR TITLE
Fix TemplateRepo no longer in models.repo

### PR DESCRIPTION
**Pull request by Caellion**
Fixes error 500 that appears when trying to browse code of a repository generated from template

### CODE REVIEW

**Comment by a1012112796** [comment]
@Caellion Hello, would you please give more detail description about this issue? Thanks

**Comment by Caellion** [comment]
> @Caellion Hello, would you please give more detail description about this issue? Thanks

Certainly.

Due to recent refactors of repo model, the template repo/header.tmpl cannot find TemplateRepo, at this line:
https://github.com/go-gitea/gitea/blob/f58e687a83b18d58298097d485f4febf98062dd7/templates/repo/header.tmpl#L41

This causes gitea to display error 500 on any repo which has been initialized from a template.

File: models/repo/repo.go [hunk 1]
```diff
@@ -683,6 +683,16 @@ func getTemplateRepo(e db.Engine, repo *Repository) (*Repository, error) {
 	return getRepositoryByID(e, repo.TemplateID)
 }
 
+// TemplateRepo returns the repository, which is template of this repository
+func (repo *Repository) TemplateRepo() *Repository {
+	repos, err := GetTemplateRepo(repo)
```
Comment (lunny) [inline models/repo/repo.go]:
```suggestion
	repo, err := GetTemplateRepo(repo)
```

Comment (lunny) [reply models/repo/repo.go]: Otherwise LGTM

**Review by zeripath — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work.

**Pull request merged**